packages/eslint-plugin/src/util/hasOverloadSignatures.ts

```diff
@@ -0,0 +1,88 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import type { RuleContext } from '@typescript-eslint/utils/ts-eslint';
+
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import type { MethodDefinitionWithBody } from './isMethodDefinitionWithBody';
+
+import { getStaticMemberAccessValue } from './misc';
+
+/**
+ * @return `true` if the function or method node has overload signatures.
+ */
+export function hasOverloadSignatures(
```

Comment:
I've extracted this a separate utility because I think it may be useful in other places (e.g. #4586).